libs/pika/runtime/src/runtime.cpp

```diff
@@ -905,15 +906,8 @@ namespace pika::detail {
             strm << std::string(79, '*') << '\n';
 
 #if defined(PIKA_HAVE_MPI)
-            int mpi_initialized = 0;
-            if (MPI_Initialized(&mpi_initialized) == MPI_SUCCESS && mpi_initialized)
-            {
-                int rank = 0;
-                if (MPI_Comm_rank(MPI_COMM_WORLD, &rank) == MPI_SUCCESS)
-                {
-                    strm << "MPI rank: " << rank << '\n';
-                }
-            }
+            if (pika::mpi::environment::is_mpi_inititialized())
+                strm << "{mpi-rank}: " << pika::mpi::environment::rank() << '\n';
```

Comment:
The MPI rank is currently (on main) printed as:
```
*******************************************************************************
MPI rank: 0
   0: PU L#0(P#0), Core L#0(P#2), Socket L#0(P#56), on pool "default"
   1: PU L#1(P#1), Core L#1(P#4), Socket L#0(P#56), on pool "default"
   2: PU L#2(P#2), Core L#2(P#5), Socket L#0(P#56), on pool "default"
   3: PU L#3(P#3), Core L#3(P#6), Socket L#0(P#56), on pool "default"
```

With this change it'd be:
```
*******************************************************************************
{mpi-rank}: 0
   0: PU L#0(P#0), Core L#0(P#2), Socket L#0(P#56), on pool "default"
   1: PU L#1(P#1), Core L#1(P#4), Socket L#0(P#56), on pool "default"
   2: PU L#2(P#2), Core L#2(P#5), Socket L#0(P#56), on pool "default"
   3: PU L#3(P#3), Core L#3(P#6), Socket L#0(P#56), on pool "default"
```
which to me looks odd compared to the output of the thread bindings. Can we keep the old style or use something that's a bit more in line with the thread bindings style?